src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionTests.cs

```diff
@@ -193,8 +193,25 @@ public async Task Dispose_FinishesSession()
         sessionToken.Value.IsCancellationRequested.Should().BeTrue();
     }
 
+    [TestMethod]
+    public void Dispose_InvokesOnSessionFailedAsync()
+    {
+        var testScope = new TestScope();
+        testScope.SetUpSwitchToBackgroundThread();
+        testScope.TestSubject.Dispose();
+
+        testScope.OnSessionFailedAsyncMock.Invocations.Should().HaveCount(1);
+    }
+
     public interface IDummyServerEvent : IServerEvent { }
 
+    private static Mock<OnSessionFailedAsync> GetOnSessionFailedAsyncMock()
```

Comment:
Not sure if this is the right way to do it but its the only way I could figure out :)

Reply:
It looks like the setup is a no-op; it would be needed if the mock is being created using `strict`.